### PR TITLE
LPS-163641 Allow configuring whether account domains restrict membership

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-163641](https://issues.liferay.com/browse/LPS-163641).

One additional question open for discussion:

- [ ] When the toggle goes from `Not restricted` to `Restricted`, should we remove membership of any user who does not have a valid email domain? Currently that is not the case, so users will be left as members after the switch.

/cc @pei-jung @ChrisKian @patriciajeanperez @david-truong @gislayne-vitorino @RafaelCRC

Screen Shot 2022-10-05 at 11 21 50 AM
Screen Shot 2022-10-05 at 11 22 07 AM

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:reevaluate:1287401_4666

**Comment by patriciajeanperez** [comment]
@drewbrokke upgrades test caught this database error:

`java.sql.SQLSyntaxErrorException: Unknown column 'accountent0_.restrictMembership' in 'field list'`

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/dao/search/AssignableAccountUserDisplaySearchContainerFactory.java [hunk 1]
```diff
@@ -108,21 +110,34 @@ else if (navigation.equals("no-assigned-account")) {
...
-
-		searchContainer.setResultsAndTotal(
-			() -> TransformUtil.transform(
-				baseModelSearchResult.getBaseModels(), AccountUserDisplay::of),
-			baseModelSearchResult.getLength());
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		try {
+			PermissionThreadLocal.setPermissionChecker(null);
```
Comment (pei-jung) [inline modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/dao/search/AssignableAccountUserDisplaySearchContainerFactory.java:117]: I'm not sure about this change. If we un-permission the user search, we are allowing users that have the `AccountEntry:Manage User` to view all the company users.

**Comment by pei-jung** [comment]
@drewbrokke we need an upgrade process to add the new column for AccountEntry

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/dao/search/AssignableAccountUserDisplaySearchContainerFactory.java (hunk 1 above)
Comment (drewbrokke) [reply modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/dao/search/AssignableAccountUserDisplaySearchContainerFactory.java:117]: I'm open to suggestions on this one. It's a similar technique that is used for assigning site members in [SelectUsersDisplayContext](https://github.com/liferay/liferay-portal/blob/b708db5d9946c5a91a7973c16958c06bd4ffa9f9/modules/apps/site/site-memberships-web/src/main/java/com/liferay/site/memberships/web/internal/display/context/SelectUsersDisplayContext.java#L214-L220).

Comment (drewbrokke) [reply modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/dao/search/AssignableAccountUserDisplaySearchContainerFactory.java]: I've removed it for now. We'll decide our direction forward after speaking with Jeff and Dave.

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:forward

**Comment by drewbrokke** [comment]
@patriciajeanperez is this failure related?

**Pull request closed without merging**